### PR TITLE
markdown workflow: add optional spell checker

### LINKED ISSUE #43: [Feature] support spell checking in markdown workflow
### Is your feature request related to a problem? Please describe the problem.

I am frustrated when I spend a lot of time to spell check my documents, and time after time, I still find issues in there :(

### Describe the solution you'd like

An option spell checker as part of markdown workflow

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by chicco785**
## Description

add optional spell checker to markdown workflow

## Changes Made

* add spell checker boolean flag
* execute spell check if flag true
* add configuration file and empty dictionary
* refactor workflows to use ternary operator to assign variables
* fix pr-label-checker

## Related Issues

Fixes #43 

## Checklist

- [x] I have used a PR title that is descriptive enough for a release note.
- [x] I have tested these changes locally.
- [x] I have added appropriate tests or updated existing tests.
- [ ] I have tested these changes on a dedicated VM or a customer VM [name of the VM]
- [ ] I have added appropriate documentation or updated existing documentation.

### CODE REVIEW

**Comment by chicco785** [comment]
to be rebased after merge of #42

File: .github/workflows/markdown.yaml [hunk 1]
```diff
@@ -92,3 +96,14 @@ jobs:
         config-file: '.github/config/md-link-config.json'
         use-verbose-mode: 'yes'
         file-extension: '.MD'
+  markdown-spellchecker:
+    if: ${{ (github.event_name == ' workflow_call' && inputs.enable-spell-check) || github.event_name != ' workflow_call' }}
```
Comment (cosimomeli) [inline .github/workflows/markdown.yaml:100]: Is the whitespace in `' workflow_call'` intentional?

Comment (chicco785) [reply .github/workflows/markdown.yaml:100]: of course not :) good catch!

Comment (chicco785) [inline .github/workflows/markdown.yaml:100]:
```suggestion
    if: ${{ (github.event_name == 'workflow_call' && inputs.enable-spell-check) || github.event_name != 'workflow_call' }}
```

**Review by cosimomeli — APPROVED** [review]

**Pull request merged**